model/device.go

```diff
@@ -13,182 +15,14 @@ func NewHardware(d *Device) *Hardware {
 	return &Hardware{Device: d, UpdatesAvailable: -1}
 }
 
+// NewDevice returns an ironlib Device object
 func NewDevice() *Device {
-	return &Device{
-		BMC:                &BMC{},
-		BIOS:               &BIOS{},
-		Mainboard:          &Mainboard{},
-		TPM:                &TPM{},
-		CPLD:               &CPLD{},
-		PSUs:               []*PSU{},
-		NICs:               []*NIC{},
-		GPUs:               []*GPU{},
-		CPUs:               []*CPU{},
-		Memory:             []*Memory{},
-		Drives:             []*Drive{},
-		StorageControllers: []*StorageController{},
-	}
+	return &Device{common.NewDevice(), nil}
 }
 
+// Device is an ironlib device object which extends the common.Device
```

Comment:
It doesn't seem that we're extending the ironlib device object anymore. Shouldn't we get rid of the `ironlib.Device` entirely and just use the bmclib one directly?

Reply:
The `ironlib.Device` object is being extended to include `OemComponents` https://github.com/metal-toolbox/ironlib/blob/61b187384a6779ad0141df6861cffebe4051d238/model/device.go#L27

Reply:
ah, I clearly missed that.